lib/Common/CommonPal.h

```diff
@@ -611,19 +611,20 @@ HRESULT ULongMult(ULONG ulMultiplicand, ULONG ulMultiplier, ULONG* pulResult);
 template <class TryFunc, class FinallyFunc>
 void TryFinally(const TryFunc& tryFunc, const FinallyFunc& finallyFunc)
 {
-    bool hasException = true;
-    try
+    class FinallyObject
     {
-        tryFunc();
-        hasException = false;
-    }
-    catch(...)
-    {
-        finallyFunc(hasException);
-        throw;
-    }
-
-    finallyFunc(hasException);
+    public:
+        FinallyObject(const FinallyFunc& finallyFunc) : finallyFunc(finallyFunc), abnormalTermination(true) {}
+        ~FinallyObject() { finallyFunc(abnormalTermination); }
+
+        void SetHasNoAbnormalTermination() { abnormalTermination = false; }
+    private:
+        const FinallyFunc& finallyFunc;
+        bool abnormalTermination;
+    } finallyObject(finallyFunc);
```

Comment:
>finallyObject(finallyFunc); [](start = 6, length = 27)

Will it run correctly if allocate finallyObject on the stack triggered stackoverflow?